main.cpp

```diff
@@ -1,10 +1,58 @@
 #include <iostream>
+#include <iomanip>   // format manipulation
 
 using namespace std;
-
+/**This program displays an equilateral triangle based on the user's input.
+ * The user decides on its size and the character that it may consist of.
+ * The triangle increases by three characters each time and is based on the 
+ * T(n)=2n+1 equation.*/
+void inputCheck(int & size)//Still trying to figure our cin.fail() and cin.good()
+{
+ bool isdigit();
+    if(size.isdigit()==true)
+        
+        cin.good();
+    cout<<endl;
+        
+    else
+        cin.fail();
+        cout << "Please re-enter a size. ";
+        cin.clear();
+        cin>>size;
+}
+void generateEquilateralTriangle(char character, int size)
+{
+     for (int i = 0; i < size; i++)
+    {
+        
+        for (int j = 0; j < (2*size+1); j++)
+        {
+            
+            if ((size+i) >= j && (size-i) <= j)
```

Comment:
So, the last time we spoke about a statement like this, it was slightly different to this one. If I can take I guess about what your statement is saying it would be,"The output of the statement would be, if the size plus 1, is more than or equal to j and the size minus  1 is less than or equal to j, the character (given by user) will be placed here, else (indicated by ":"), the space will be output if the  firstrequirement is not met.". Do I understand this correctly? The "?" means, "if this statement is true"?